File: backend/app/models/result.py
from pydantic import BaseModel
from typing import Dict, List, Optional

class ResultModel(BaseModel):
    id: Optional[int] = None

    image_urls: List[str] = []
    audio_urls: List[str] = []
    text: Optional[str] = None
    processed_texts: List[Dict] = []
    
    img_label_idx: Optional[int] = None
    img_label_name: Optional[str] = None
    img_prob: Optional[float] = None

    img_txt_label_idx: Optional[int] = None
    img_txt_label_name: Optional[str] = None
    img_txt_prob: Optional[float] = None
    img_txt_weights: List[List[float]] = []

    full_label_idx: Optional[int] = None
    full_label_name: Optional[str] = None
    full_prob: Optional[float] = None
    full_weights: List[List[float]] = []
    
    class Config:
        from_attributes = True

File: frontend/src/app/types/result.ts
export type PredictionType = "img" | "img-txt" | "full"

export interface PredictionResult {
  labelName: string;
  prob: number;
  weights?: number[][];
}

export interface UploadRes {
  id: number;
}

interface Word {
  word: string;
  sentence: string;
}

export interface ResultRes {
  id: number;
  text: string;
  processedTexts: Word[];
  imageUrls: string[];
  audioUrls: string[];
  imgLabelIdx: number;
  imgLabelName: string;
  imgProb: number;
  imgTxtLabelIdx: number;
  imgTxtLabelName: string;
  imgTxtProb: number;
  imgTxtWeights: number[][];
  fullLabelIdx: string;
  fullLabelName: string;
  fullProb: number;
  fullWeights: number[][];
}
